src/industry_cmd.cpp

```diff
@@ -1156,15 +1177,20 @@ static void ProduceIndustryGoods(Industry *i)
 
 	i->counter--;
 
-	/* produce some cargo */
+	/* If using an industry callback, scale the callback interval by cargo scale percentage. */
+	if (HasBit(indsp->callback_mask, CBM_IND_PRODUCTION_256_TICKS)) {
+		if (i->counter % ScaleByInverseCargoScale(Ticks::INDUSTRY_PRODUCE_TICKS, false) == 0) {
+			IndustryProductionCallback(i, 1);
+			ProduceIndustryGoodsHelper(i, false);
+		}
+	}
+
+	/* All other production and special effects happen every 256 ticks, and cargo production is just scaled by the cargo scale percentage. */
```

Comment:
Why two different methods of scaling anyway?

Reply:
We could scale the interval for all cargos, but that makes the production quite jumpy at low scales -- vehicles sit empty for a long time waiting for the first production ticks. It annoys me in the JGRPP implementation of daylength, and I'd rather avoid it if possible.